### PR TITLE
parse out leading exclamations after <at>botname

### PR DESCRIPTION
**Pull request by crclark96**
parses out leading `!` after `@botname`, issue #157

### CODE REVIEW

**Comment by crclark96** [comment]
screenshot 2019-02-09 11 20 19

File: server/botserver.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ def parse_slack_message(self, message_list):
             if msg.get("type") == "message" and "subtype" not in msg:
                 if self.bot_at in msg.get("text", ""):
                     # Return text after the @ mention, whitespace removed
-                    return msg['text'].split(self.bot_at)[1].strip(), msg['channel'], msg['thread_ts'] if 'thread_ts' in msg else msg['ts'], msg['user']
+                    # leading ! removed
```
Comment (Grazfather) [inline server/botserver.py]: looks good, just combine this with the comment one line above

Comment (crclark96) [reply server/botserver.py]: PEP8 requires 79 character max line length, but that's going to be a whole thing

updated

**Comment by crclark96** [comment]
updated

**Comment by Kileak** [comment]
Nice, less bishop spamming :)

**Comment by Kileak** [comment]
After giving it a try, I don't see any value in this change, since it will only strip an exclamation mark that is following the bot name directly.

Not sure, if many people really try to talk to bishop this way.

Any other constellation like

sup @bishop what's uuuup?!

will still result in an error message (which is totally fine from my point of view)

**Comment by Kileak** [comment]
Closing the pr after talking to @crclark96

**Pull request closed without merging**